packages/sanity/src/core/releases/tool/detail/documentTable/extractSearchableText.ts

```diff
@@ -0,0 +1,52 @@
+import {toPlainText} from '@portabletext/react'
+import {isPortableTextBlock} from '@portabletext/toolkit'
+
+/**
+ * Extracts a searchable string from a field value, handling various data types
+ * @param value - The field value to extract text from
+ * @returns A string representation suitable for search
+ */
+export function extractSearchableText(value: unknown): string {
+  if (typeof value === 'undefined' || value === null) return ''
+
+  // Handle primitive types
+  if (typeof value === 'string') return value
+  if (typeof value === 'number') return value.toString()
+
+  // Handle arrays
+  if (Array.isArray(value)) {
+    // Check if it's portable text
+    if (value.length > 0 && isPortableTextBlock(value[0])) {
+      return toPlainText(value)
+    }
+    // Handle regular arrays (join with space)
+    return value.map(extractSearchableText).filter(Boolean).join(' ')
```

Comment:
Do we want it to recurse over the entire value/document? For large documents this could have a significant perf impact. I don't think we need to go deeper than a couple of levels.